src/averageAges.js

```diff
@@ -1,61 +1,51 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const result = people.filter(person => person.sex === 'm'
+    && (!century || Math.ceil(person.died / 100) === century));
+  const finalResult = result.reduce((sum, person) =>
+    sum + person.died - person.born, 0);
+
+  return finalResult / result.length;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const result = people.filter(person => person.sex === 'f'
+    && (!withChildren || people.find(children =>
+      children.mother === person.name)));
+  const finalResult = result.reduce((sum, person) =>
+    sum + person.died - person.born, 0);
+
+  return finalResult / result.length;
```

Comment:
this logic is the same as in the `calculateMenAverageAge`, so move it to a separate function